pennylane_lightning/src/simulator/DynamicDispatcher.hpp

```diff
@@ -19,68 +19,140 @@
 
```

Comment:
Really excited to make use of this class !

Reply:
Yes, but we need to think a bit more about a better dispatch mechanism if we want to add OpenMP, AVX512, GPU kernels etc.